src/ai_migrate/example_selector.py

```diff
@@ -0,0 +1,158 @@
+    "selected_examples": [
+        {
+            "id": "Example number (integer)",
+            "reason": "Detailed justification for selecting this example"
+        }
+    ],
+    "excluded_examples": [
+        {
+            "id": "Example number (integer)", 
+            "reason": "Brief reason for excluding this example"
+        }
+    ]
+}
+
+Notes:
+- Example numbers should be integers (1, 2, 3, etc.)
+- Provide clear, specific reasons for each selection and exclusion
+- Focus on selecting examples that demonstrate patterns needed for this specific migration
+- Ensure the response is valid JSON that can be parsed programmatically"""
+
+
+@dataclass
+class ExampleSelectionResult:
+    selected_examples: List[MigrationExample]
+    analysis: str
+    selection_reasons: dict[str, str]
+    exclusion_reasons: dict[str, str]
+
+
+async def select_relevant_examples(
+    target_files: List[str],
+    available_examples: List[MigrationExample],
+    client: DefaultClient,
+) -> ExampleSelectionResult:
+    target_content = []
+    for target_file in target_files:
+        path = Path(target_file)
+        content = path.read_text()
+        target_content.append(f"### `{path.name}`\n```\n{content}\n```")
+
+    examples_content = []
+    for i, example in enumerate(available_examples, 1):
+        example_files = []
+        for old_file in example.old_files:
+            example_files.append(f"### `{old_file.name}`\n```\n{old_file.content}\n```")
+            for new_file in example.new_files:
+                if new_file.name == old_file.name:
+                    example_files.append(
+                        f"### `{new_file.name}` (migrated)\n```\n{new_file.content}\n```"
+                    )
+                    break
+        examples_content.append(
+            f"Example {i} ({example.name}):\n" + "\n".join(example_files)
+        )
+
+    prompt = (
+        f"{EXAMPLE_SELECTION_PROMPT}\n\n"
+        f"Target Files:\n{chr(10).join(target_content)}\n\n"
+        f"Available Examples:\n\n{chr(10).join(examples_content)}"
+    )
+
+    messages = [
+        {"role": "system", "content": EXAMPLE_SELECTION_PROMPT},
+        {"role": "user", "content": prompt},
+    ]
+
+    response, _ = await client.generate_completion(messages=messages, temperature=0.1)
+
+    try:
+        if "choices" in response and response["choices"]:
+            message = response["choices"][0].get("message", {})
+            if isinstance(message, dict):
+                content = message.get("content", "")
+            else:
+                content = str(message)
+        else:
+            content = str(response)
+
+        if not content.strip():
+            content = json.dumps(
+                {
+                    "analysis": "No analysis provided - LLM returned empty response",
+                    "selected_examples": [],
+                    "excluded_examples": [],
+                }
+            )
+    except Exception as e:
```

Comment:
avoid catching Exception, the stack trace should have enough info here

Reply:
ack